Fixes #17599 - Smart vars allow you to select 'none' class

dLobatog opened the pull request:
When you edit a smart variable, it allows in the form to 'not' associate
it with any Puppet class. This doesn't show any error on the UI, however
it's not associated and you will see "Failed to save: Puppetclass can't
be blank" on the logs.

Additionally some strings were not translated at all so I took the
chance to add support for that.

domcleal commented on app/views/lookup_keys/_fields.html.erb:
```diff
@@ -4,9 +4,9 @@
   <fieldset>
     <h2><%= _("Parameter details") %></h2>
-    <%= text_f f, :key, :disabled => is_param, :size => "col-md-8" %>
+    <%= text_f f, :key, :disabled => is_param, :size => "col-md-8", :label => _('Key')  %>
```
Note sure that model attributes need extracting, they're already in model_attributes.rb (but probably under the wrong name as 6b4b489 seemed to rename them).

This will mean twice the number of translations, but only one gets used. Better fix this separately if it's really an issue, it's unrelated.

domcleal reviewed (changes requested):
Best to remove the unrelated i18n changes and file a ticket with specifics about the issue.

dLobatog committed:
Fixes #17599 - Smart vars allow you to select 'none' class

When you edit a smart variable, it allows in the form to 'not' associate
it with any Puppet class. This doesn't show any error on the UI, however
it's not associated and you will see "Failed to save: Puppetclass can't
be blank" on the logs.

Additionally some strings were not translated at all so I took the
chance to add support for that.

dLobatog commented:
Updated to remove translations - they're on https://github.com/theforeman/foreman/pull/4097 now

domcleal commented:
Merged with the i18n references removed from the commit message. Thanks @dLobatog.
